# MK-Fst-Phylo

## Description

Pipeline to calculate Fst distance for each pair of samples. Makes Bootstrap or sub-sample technique to make many Fst distance inferences. The Fst distances are used then to make Neighbor-Joining phylogenetic inference and finally a consensus phylogenetic tree is made with all the replicates (subsamples or bootstrap).

The pipeline takes as input multi-sample (or population) BCF files or PLINK format files (bed, fam and bim).

The pipeline is divided in six modules:

 - **000-BCF_to_BED**: This module converts a multi-sample BCF file into binary files used by software PLINK (bed, fam and bim).
 
 - **000b-BCFBootsBED**: Module for generates bootstrap BED files  taking as input a BCF file.
 
 - **001-SubSampleFst**:  Calculates Fst values for each pair of samples in an input PLINK format files (bed, fam, bim). For each Fst inference is performed quasi bootstrap (formally sampling), taking without replacement the 10% of genetic variants.
 
 - **001b-BoostrapFst**: Calculates Fst values for each pair of samples in an input PLINK format files (bed, fam, bim). For each Fst inference is performed formal bootstrap, taking with replacement the same number of genetic variants as the input.
 
 - **001c-Fst**: This module takes as input PLINK format files (bed, fam, bim) and calculates Fst values for each pair of samples. Ideally used this module taking as input the module *000b-BCFBootsBED*.
 
 - **002-NeighbourPhylip**: Construct Neighbor-Joining phylogeny trees in  [newick format](http://evolution.genetics.washington.edu/phylip/newicktree.html), taking as input distance matrix. Can receive as input either the results for module *001-SubSampleFst*, *001b-BoostrapFst* or *001c-Fst*.
 
 - **003-ConsensusTree**: The module construct a Consensus tree in [newick format](http://evolution.genetics.washington.edu/phylip/newicktree.html), taking as input many newick tress.
 

## Pipeline configuration

### Input BCF

The input BCF  should  be generated by BCFtools ideally version 1.5.

### Config file

Create a config file named *config.mk* in the *analysis* directory that contain the next variables:

 - **bots**:  Number of Bootstrap or sub-sampling replicates to perform.
 - **NT**: Number of threads to use.
 - **OUT**: Sample out-group number (according to the bim file) to be used to root phylogeny trees.
 - **per**: Percentage of data to use for each sub-sampling replicate.
### Software dependencies 

- [BCFtools](https://samtools.github.io/bcftools/bcftools.html) v1.5
- [consense](http://evolution.genetics.washington.edu/phylip/doc/consense.html) *
- [neighbor](http://evolution.genetics.washington.edu/phylip/doc/neighbor.html) *
- [PLINK](http://www.cog-genomics.org/plink2) v1.9
- [Eigensoft](https://github.com/DReichLab/EIG) v1.9




**NOTE: Those programs comes with the package [PHYLIP](http://evolution.genetics.washington.edu/phylip/) v. 3.6 .*




## Output data format
The output from module 003, will be a consensus tree that integrates bootstrap values. Consensus tree in [newick format](http://evolution.genetics.washington.edu/phylip/newicktree.html). You can visualize that kind of files in a software as [MEGA](https://www.megasoftware.net/)
. An example of the generated tree is see below:

![Phylogenetic tree](https://github.com/FRPV/planets/blob/master/Tree.png)


## Additional notes

 - This pipeline was used to calculate Phylogeny in the 100G-MX samples.

## References

[1] Felsenstein, J. (1993). *_PHYLIP (phylogeny inference package), version 3.5 c_*. Joseph Felsenstein.

[2] Price, A. L., Patterson, N. J., Plenge, R. M., Weinblatt, M. E., Shadick, N. A., & Reich, D. (2006). *Principal components analysis corrects for stratification in genome-wide association studies.* _Nature genetics_, _38_(8), 904.

[3] Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A., Bender, D., ... & Sham, P. C. (2007). PLINK: a tool set for whole-genome association and population-based linkage analyses. _The American Journal of Human Genetics_, _81_(3), 559-575. [https://doi.org/10.1086/519795](https://doi.org/10.1086/519795 "Persistent link using digital object identifier")

## Author Info
Developed by: [Fernando Perez](https://www.linkedin.com/in/fernandorpv/)

email: [fperez@inmegen.gob.mx](mailto:fperez@inmegen.gob.mx)

